src/backend/app/auth/roles.py

```diff
@@ -134,6 +134,42 @@ async def org_admin(
     return user_data
 
 
+async def project_admin(
+    project: DbProject = Depends(get_project_by_id),
+    db: Session = Depends(get_db),
+    user_data: AuthUser = Depends(login_required),
+):
+    """Project admin role."""
+    user_id = await get_uid(user_data)
+
+    if await check_org_admin(
+        db, user_data, project, project.organisation_id
+    ) or await check_super_admin(db, user_data):
```

Comment:
I'm starting to see that the current db schema we have makes for a very inefficient check for roles.

If the role check has to run the check for each role in sequence, we will have up to 5 database calls...
We may need to rethink - I really haven't spent as much time on roles as I could have recently.

A way to alleviate this for the current implementation is to check roles in inverse order.
mapper --> project admin --> org_admin --> super_admin

The reason I suggest this is that we probably have a pyramid of users, where most will be mappers, and very few will be org admins or super admins.

We can optimise so the performance is better for mappers, then is progressively slower for users higher in the hierarchy.

Reply:
One thing i noticed just now is Since we are asigning user in a table organisation_managers as an org_admin, and the same user may have role as a mapper in userroles table. I think we should think about this to check the roles from lowest hierarchy.

Reply:
If the user has a mapper roles for a specific project, but is the organisation admin for the org, then the mapper role will be ignored. I think we should always use the highest level of access the user has.

Is this behaviour ok, or am I missing something?

Reply:
yeah it is ok. These new changes addresses it.